app/operations/fdsh/ssa/h3/request_json_ssa_verification.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Fdsh
+  module Ssa
+    module H3
+      # This class takes a json representing a person as input and invokes SSA.
+      class RequestJsonSsaVerification
+        include Dry::Monads[:result, :do, :try]
+        include EventSource::Command
+
+        PublishEventStruct = Struct.new(:name, :payload, :headers)
+
+        PUBLISH_EVENT = "verify_ssa_composite_service_rest_request"
+        # @param params [String] the json payload of the person
+        # @return [Dry::Monads::Result]
+        def call(params)
+          validated_params = yield validate_params(params)
+          publish_event(validated_params)
+        end
+
+        protected
+
+        def validate_params(params)
+          return Failure('Cannot publish payload without transmittable objects') unless params[:transmittable_objects]
+          unless params[:transmittable_objects][:transaction].is_a?(::Transmittable::Transaction)
+            return Failure('Cannot publish payload without transaction')
+          end
+          unless params[:transmittable_objects][:transmission].is_a?(::Transmittable::Transmission)
+            return Failure('Cannot publish payload without transaction')
+          end
+          return Failure('Cannot publish payload without transaction') unless params[:transmittable_objects][:job].is_a?(::Transmittable::Job)
+          return Failure('Cannot publish payload without correlation_id') unless params[:correlation_id].is_a?(String)
+          return Failure('Cannot publish payload without jwt token') unless params[:token].is_a?(String)
+          return Failure('Cannot publish payload without transaction json payload') unless params[:transmittable_objects][:transaction].json_payload
+          return Failure('Cannot publish payload without message id') unless params[:transmittable_objects][:job].message_id
+
+          Success(params)
+        end
+
+        def publish_event(params)
+          event = PublishEventStruct.new(PUBLISH_EVENT, params[:transmittable_objects][:transaction].json_payload,
+                                         { authorization: "Bearer #{params[:token]}",
+                                           messageid: params[:transmittable_objects][:job].message_id,
+                                           partnerid: ENV['CMS_PARTNER_ID'] })
+          result = update_status(params[:transmittable_objects])
```

Comment:
can we move this check before instantiating the event?

Reply:
We were trying to follow the standard practice of updating the states after we save the objects. However, in this case, it is like an OpenStruct. But wanted to treat it like a regular object. But if you strongly feel that we should do it before let us know we can do that.